browser/js/common/directives/reviews/reviews.js

```diff
@@ -0,0 +1,62 @@
+app.directive('reviews', function ($rootScope, AuthService, AUTH_EVENTS, $state, ReviewFactory, ProductFactory) {
+
+    return {
+        restrict: 'E',
+        scope: {},
+        templateUrl: 'js/common/directives/reviews/reviews.html',
+        link: function (scope, elem, attrs) {
+
+            scope.alreadySubmitted = false;
+            scope.isLoggedIn = AuthService.isAuthenticated;
+
+            // submitting a new review.
+            // get loggedIn User to append to review.
+            scope.submit = function(){
+                if (AuthService.isAuthenticated()) {
+                    AuthService.getLoggedInUser()
+                        .then(function(user){
+                            // create new review in database
+                            return ReviewFactory.createReview({
+                                user_id: user._id,
+                                product_id: attrs.product,
+                                text: scope.review.text,
+                                rating: scope.review.rating
+                            })
+                        }).then(function(newReview){
+                            // update the view with new review
+                            scope.reviews.push(newReview);
+                            scope.alreadySubmitted = true;
+                            scope.submittedMessage = "Thank you for submitting a review.";
+                        })
+                } else {
+                    alert("You must be logged in to create reviews.");
+                    //ReviewFactory.createReview({
+                    //    user_id: "123",
+                    //    product_id: attrs.product,
+                    //    text: scope.review.text,
+                    //    rating: scope.review.rating
+                    //})
+                }
+                scope.newReviewForm.$setPristine;
+            }
+
+            // get Product from product id that was passed into directive
+            ProductFactory.fetchById(attrs.product)
+                .then(function(product){
```

Comment:
When this directive is used on a product page, the product state will have the full product object on the scope. So you may not need to do this. I'm not sure about for the user page. What if we just add the product name to our review model?